tensorflow_datasets/core/dataset_info.py

```diff
@@ -408,19 +409,52 @@ def initialize_from_bucket(self):
       gcs_utils.download_gcs_file(fname, out_fname)
     self.read_from_directory(tmp_dir)
 
-  def __str__(self):
-    splits_pprint = "{\n %s\n    }" % (
-        pprint.pformat(
-            {k: self.splits[k] for k in sorted(list(self.splits.keys()))},
-            indent=8, width=1)[1:-1])
-    features_dict = self.features
-    features_pprint = "%s({\n %s\n    }" % (
-        type(features_dict).__name__,
-        pprint.pformat({
-            k: features_dict[k] for k in sorted(list(features_dict.keys()))
-        }, indent=8, width=1)[1:-1])
-    citation_pprint = '"""\n%s\n    """' % "\n".join(
-        [u" " * 8 + line for line in self.citation.split(u"\n")])
+  def show_examples(self, ds, rows=3, cols=3, plot_scale=3):
+    """Display some random (rows*columns) images from the image-dataset
+    Args:
+      ds: `tf.data.Dataset`. Does not accept batch data.
+      rows: `int`, number of rows of the display grid.
+      cols: `int`, number of columns of the display grid.
+    """
+    image_keys = [
+        k for k, feature in self.features.items()
+        if isinstance(feature, features_lib.Image)
+    ]
+    # TODO: Autodetect ClassLabel if present recursively.
+    label_keys = [
+        k for k, feature in self.features.items()
+        if isinstance(feature, features_lib.ClassLabel)
+    ]
+    
+    if len(image_keys) != 1:
+      raise ValueError(
+          'Visualisation not supported for dataset `{}`. Was not able to auto-infer '
+          'image and label keys.'.format(self.name))
+    else:
+      image_key = image_keys[0]
+      num_examples = rows * cols
+      examples = list(dataset_utils.as_numpy(ds.take(num_examples)))
+      fig = plt.figure(figsize=(plot_scale*cols, plot_scale*rows))
+      fig.subplots_adjust(hspace=1/plot_scale, wspace=1/plot_scale)
+      for i, ex in enumerate(examples):
+        fig.add_subplot(rows, cols, i+1)
+        image = ex[image_key]
+        if image.shape[2] == 1:
+          image = image.reshape(image.shape[:2])
+        plt.imshow(image, cmap='gray')
+        if len(label_keys) == 1:
+          label_key = label_keys[0]
+          label = ex[label_key]
+          plt.xlabel("label: int={}, str={}".format(label, self.features[label_key].int2str(label)))
+      plt.show()
```

Comment:
Maybe also return plt, so that users can save the plot afterwards.

Reply:
Is this necessary, actually when it is displayed we get an option to save the plot manually.

Reply:
I would prefer. It does not cost anything and some people may prefer controlling the plot from their python code rather than UI